graphql/data/schema.js

```diff
@@ -2392,14 +2579,13 @@ const queryType = new GraphQLObjectType({
       type: userImpactType,
       args: {
         userId: { type: new GraphQLNonNull(GraphQLString) },
+        // currently still taking charity id to not cause breaking changes
         charityId: { type: new GraphQLNonNull(GraphQLString) },
       },
       resolve: async (_, args, context) => {
-        const { userId, charityId } = args
-        return (await UserImpactModel.getOrCreate(context.user, {
-          userId,
-          charityId,
-        })).item
+        const { userId } = args
+        const userImpact = await getUserImpact(context.user, userId)
+        return userImpact
```

Comment:
Let's leave this as-is. Instead, add an impact resolver to the user, like `currentImpact`. You're right that this shouldn't be a root field. Now that it is, though, I think it should be able to resolve to any item in UserImpact.